services/QuillLMS/app/controllers/diagnostic_insights_controller.rb

```diff
@@ -17,6 +19,20 @@ def diagnostics_assigned
     render json: fetch_diagnostics_assigned.to_json(root: false)
   end
 
+  def complete_history
+    # TODO: make this cache smarter or eradicate the need for it by speeding up the query
+    complete_history_by_classroom_data = Rails.cache.fetch("#{COMPLETE_HISTORY_CACHE_KEY}.#{current_user.id}", expires_in: 24.hours) do
```

Comment:
I think it's worth using the `UserCacheable` concern for cache key stuff.  It has a bunch of nice built-in cache expiration logic when students complete activities so we don't have to just wait 24 hours.
```ruby
classroom = Classroom.find(classroom_id)
current_user.all_classrooms_cache(key: 'diagnostifc_insights.complete_history') do
  ...
end
```